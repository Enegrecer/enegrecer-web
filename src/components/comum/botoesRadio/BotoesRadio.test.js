import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import BotoesRadio from './index';

describe('Botões Rádio', () => {
  const botoes = [
    { id: 'radioInjuria', valor: 'injuria', label: 'Injúria' },
    { id: 'radioRacismo', valor: 'racismo', label: 'Racismo' }];

  it('deve possuir estrutura definida ', () => {
    const onChange = jest.fn();
    const botoesRadio = mount(<BotoesRadio
      classes={'col s3'}
      id={'id'}
      onChange={onChange}
      botoes={botoes}
    />);
    const botoesRadioTextoJson = toJson(botoesRadio);
    expect(botoesRadioTextoJson).toMatchSnapshot();
  });
});

