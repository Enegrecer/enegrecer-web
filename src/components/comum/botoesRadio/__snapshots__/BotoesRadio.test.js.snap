// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Botões Rádio deve possuir estrutura definida  1`] = `
<botoesRadio
  botoes={
    Array [
      Object {
        "id": "radioInjuria",
        "label": "Injúria",
        "valor": "injuria",
      },
      Object {
        "id": "radioRacismo",
        "label": "Racismo",
        "valor": "racismo",
      },
    ]
  }
  classes="col s3"
  id="id"
  onChange={[Function]}
>
  <div
    className="col s3"
  >
    <p
      key="radioInjuria"
    >
      <input
        id="radioInjuria"
        name="id"
        type="radio"
      />
      <label
        htmlFor="radioInjuria"
      >
        injuria
      </label>
    </p>
    <p
      key="radioRacismo"
    >
      <input
        id="radioRacismo"
        name="id"
        type="radio"
      />
      <label
        htmlFor="radioRacismo"
      >
        racismo
      </label>
    </p>
  </div>
</botoesRadio>
`;
